client/ayon_nuke/api/colorspace.py

```diff
@@ -0,0 +1,334 @@
+    over all possible combinations of display and view colorspaces. Those
+    could be separated by COLOR_VALUE_SEPARATOR defined in constants.
+
+    If Anatomy template tokens are used but formatting data is not provided,
+    it will try any other available variants in next position of the separator.
+
+    This method also validate that the formatted display and view profile is
+    available in currently run nuke session ocio config.
+
+    Example:
+        >>> from ayon_nuke.api.colorspace import get_formatted_display_and_view_as_dict  # noqa
+        >>> view_profile = {
+        ...     "view": "{context};sRGB",
+        ...     "display": "{project_code};ACES"
+        ... }
+        >>> formatting_data = {
+        ...    "context": "01sh010",
+        ...    "project_code": "proj01"
+        ...}
+        >>> display_and_view = get_formatted_display_and_view_as_dict(
+        ...    view_profile, formatting_data)
+        >>> print(display_and_view)
+        {"view": "01sh010", "display": "proj01"}
+
+
+    Args:
+        view_profile (dict): view and display profile
+        formatting_data (dict): formatting data
+        root_node (Optional[nuke.Node]): root node
+
+    Returns:
+        dict: formatted display and view profile in dict
+            ex: {"view": "sRGB", "display": "ACES"}
+    """
+    if not root_node:
+        root_node = nuke.root()
+
+    views = view_profile["view"].split(COLOR_VALUE_SEPARATOR)
+
+    # display could be optional in case nuke_default ocio config is used
+    displays = []
+    if view_profile["display"]:
+        displays = view_profile["display"].split(COLOR_VALUE_SEPARATOR)
+
+    # generate all possible combination of display/view
+    display_views = []
+    for view in views:
+        # display could be optional in case nuke_default ocio config is used
+        if not displays:
+            display_views.append({"view": view.strip(), "display": None})
+            continue
+
+        for display in displays:
+            display_views.append(
+                {"view": view.strip(), "display": display.strip()})
+
+    for dv_item in display_views:
+        # format any template tokens used in the string
+        profile = {
+            "view": StringTemplate(dv_item["view"]).format_strict(
+                formatting_data),
+            "display": StringTemplate(dv_item["display"]).format_strict(
+                formatting_data) if dv_item["display"] else None,
+        }
+        log.debug("Resolved profile: `{}`".format(profile))
+
+        # making sure formatted colorspace exists in running session
+        # also need to test case where display is None
+        test_string = profile["view"]
+        if profile["display"]:
+            test_string = create_viewer_profile_string(
+                profile["view"], profile["display"], path_like=False
+            )
+
+        if test_string in get_display_and_view_colorspaces(root_node):
+            return profile
```

Comment:
```suggestion
    root_display_and_view = get_display_and_view_colorspaces(root_node)
    for dv_item in display_views:
        # format any template tokens used in the string
        view = StringTemplate.format_strict_template(
            dv_item["view"], formatting_data
        )
        display = dv_item["display"]
        if display:
            display = StringTemplate.format_strict_template(
                display, formatting_data
            )

        log.debug(f"Resolved View: '{view}' Display: '{display}'")

        # making sure formatted colorspace exists in running session
        # also need to test case where display is None
        test_string = view
        if display:
            test_string = create_viewer_profile_string(
                view, display, path_like=False
            )

        if test_string in root_display_and_view:
            return {
                "view": view,
                "display": display,
            }
```

Reply:
`# Make sure formatted colorspace exists in running session` is that comment now in the wrong place?

Reply:
> # Make sure formatted colorspace exists in running session is that comment now in the wrong place?

I would say it is related to `if test_string in root_display_and_view:` condition.